src/controllers/jobRoleController.ts

```diff
@@ -27,6 +27,54 @@ export class JobRoleController {
     }
   };
 
+  /**
+   * Handles job role deletion
+   * POST /job-roles/:id/delete
+   */
+  public deleteJobRole = async (req: Request, res: Response): Promise<void> => {
+    try {
+      const jobId = req.params.id;
+
+      if (!jobId) {
+        res.status(400).render("error", {
+          title: "Error",
+          message: "Job ID is required",
+          error: "Invalid job ID provided",
+        });
+        return;
+      }
+
+      // Make HTTP request to backend API to delete the job role
+      const backendUrl = `http://localhost:3001/api/jobs/${jobId}`;
+      const response = await fetch(backendUrl, {
+        method: "DELETE",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      if (response.ok) {
+        // Success - redirect back to job roles list
+        res.redirect("/job-roles?message=Job deleted successfully");
+      } else {
+        // Error from backend
+        const errorData = await response.json();
+        res.status(500).render("error", {
+          title: "Delete Failed",
+          message: "Failed to delete job role",
+          error: errorData.message || "Unknown error occurred",
+        });
```

Comment:
Controller should just call the service - service will do the communication with the API.
You'll need to update the existing interface `JobRoleService` and current in-memory implementation.